WordPress/openverse
# Content Safety UI for single results

## Fixes



Fixes #2462 
Fixes #2463 

## Description



- Adds the `VSafetyWall` for single results
- Adds the `VHideButton` for hiding single results
- Implements the VSafetyWall on single result views
- Abstracts single result "back to" and "hide" links into a unified `VSingleResultControls` component
  - Fixes the size of the "back to" button as per #2657
## Todo

- [x] Add strings to en.json instead of hardcoding
- [x] Add real copy to the `VSafetyWall` with strings dependent on the sensitive content reasons ('user_reported_sensitive', 'sensitive_text')
- [ ] Update single result e2e tests
- [x] Button aria labels and a11y refinements
- [x] Connect `useSensitiveMedia` to one of the stores, so hidden/shown media persists correctly on subsequent views (might also be able to wait for unification with other code like #2640 

## Testing Instructions


- Run the frontend locally with `j f`
- Go to `/preferences` and enable all sensitive-content related settings (`sensitive_content ` and `fake_sensitive`
- Conduct a search; click on any blurred result
- Review the behavior and styling of the single results
- Bonus: Leave a single result revealed, go back to the search and observe it is now visible. Then, disable and reenable the blur toggle in the filter sidebar. The revealed result will be hidden again

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
